calculateAverage_matei88.php

```diff
@@ -0,0 +1,161 @@
+
+    $chunks = [];
+    $chunk_start = 0;
+    while ($chunk_start < $size) {
+        $chunk_end = min($size, $chunk_start + $chunk_size);
+
+        if ($chunk_end < $size) {
+            fseek($fp, $chunk_end);
+            fgets($fp);
+            $chunk_end = ftell($fp);
+        }
+
+        $chunks[] = [
+            $chunk_start,
+            $chunk_end
+        ];
+
+        $chunk_start = $chunk_end;
+    }
+
+    fclose($fp);
+    return $chunks;
+}
+
+
+// Function to perform a task in child process
+function performTask(string $file, int $chunk_start, int $chunk_end, $processId, $sharedMemoryId): void
+{
+    $stations = [];
+    $fp = fopen($file, 'rb');
+    fseek($fp, $chunk_start);
+
+    while (($line = fgets($fp)) !== false && ftell($fp) <= $chunk_end) {
+        // Extract city and temperature
+        $city = strtok($line, ";");
+        $temp = (float) strtok(PHP_EOL);
+
+        // Update the stations array
+        if ($station = &$stations[$city]) {
+            $station[3]++;           // Increment count
+            $station[2] += $temp;    // Add temperature to sum
+            if ($temp < $station[0]) {
+                $station[0] = $temp; // Update min temperature
+            } elseif ($temp > $station[1]) {
+                $station[1] = $temp; // Update max temperature
+            }
+        } else {
+            $stations[$city] = [
+                $temp,  // Min temperature
+                $temp,  // Max temperature
+                $temp,  // Sum temperature
+                1       // Count
+            ];
+        }
+    }
+    $result = '';
+    foreach($stations as $k => $station) {
+        $result .=  $k . '#' . implode('#', $station) . PHP_EOL;
+    }
+
+    shmop_write($sharedMemoryId, $result, $processId * 1024 * 1024);
+}
+
+$chunks = get_file_chunks($file, $processCount);
+
+// Fork processes
+for ($i = 0; $i < $processCount; $i++) {
+    $pid = pcntl_fork();
+    if ($pid == -1) {
+        die("Could not fork process\n");
+    } elseif ($pid) {
+        // Parent process
+        $pids[] = $pid;
+    } else {
+        // Child process
+        performTask($file, $chunks[$i][0], $chunks[$i][1], $i, $sharedMemoryId);
+        exit(0); // Exit child process
+    }
+}
+
+// Wait for child processes and read results
+foreach ($pids as $pid) {
+    pcntl_waitpid($pid, $status);
+}
+
+//$finish_parse = microtime(true);
+
+$results = [];
+for ($i = 0; $i < $processCount; $i++) {
+    $lines = explode(PHP_EOL, shmop_read($sharedMemoryId, $i * 1024 * 1024, 1024 * 1024));
+
+    foreach ($lines as $v) {
+        $line = explode('#', $v);
+        if (empty($line[1])) {
+            continue;
+        }
+        [$city, $min, $max, $sum, $count] = $line;
+        if ($result = &$results[$city]) {
+            $result[2] += (float) $sum;
+            $result[3] += (int) $count;
+            if ($min < $result[0]) {
+                $result[0] = $min;
+            }
+            if ($max < $result[1]) {
```

Comment:
```suggestion
            if ($max > $result[1]) {
```
This was wrong in the other implementations as well, see #5